github-extension/src/common/github-service.ts

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { Repository, Collaborator, User, Credentials, PullRequest, Organization } from "./github-model";
+
+export const githubKeyServicePath = '/services/github';
+export const GithubService = Symbol("GithubService");
+
+/**
+ * Representation of JSON-RPC service for GitHub communication.
+ */
+export interface GithubService {
+
+    /**
+     * Get specified repository.
+     * 
+     * @param {string} owner Owner of the repository.
+     * @param {string} repository Name of the repository.
+     */
+    getRepository(credentials: Credentials, owner: string, repository: string): Promise<Repository>;
+
+    /**
+     * Get repositories owned by a user.
+     * 
+     * @param {string} user Owner of the repository.
+     * @param {number} [pageNumber=0] Number of the page (optional, 0 by default).
+     * @param {number} [pageSize=0] Size of the page (optional, 0 by default).
+     */
+    getUserRepositories(credentials: Credentials, user: string, pageNumber?: number, pageSize?: number): Promise<Repository[]>;
+
+    /**
+     * Get repositories owned by an organization.
+     * 
+     * @param {string} organization Organization witch repository belongs to.
+     * @param {number} [pageNumber=0] Number of the page (optional, 0 by default).
```

Comment:
FYI in typescript you can say it's optional by using ? on the parameter

Reply:
done